File: Examples/VisualBasic/Working-with-Masters/RetrieveMasterInfo.vb
Imports Microsoft.VisualBasic
Imports System.IO
Imports Aspose.Diagram
Imports System

Namespace Diagrams
    Public Class RetrieveMasterInfo
        Public Shared Sub Run()
            ' ExStart:RetrieveMasterInfo
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_Master()

            ' Call the diagram constructor to load diagram from a VDX file
            Dim vdxDiagram As New Diagram(dataDir & "RetrieveMasterInfo.vdx")

            For Each master As Aspose.Diagram.Master In vdxDiagram.Masters
                ' Display information about the masters
                Console.WriteLine(Constants.vbLf & "Master ID : " & master.ID)
                Console.WriteLine("Master Name : " & master.Name)
            Next master

            Console.ReadLine()
            ' ExEnd:RetrieveMasterInfo
        End Sub
    End Class
End Namespace

File: Examples/CSharp/Working-with-Masters/RetrieveMasterInfo.cs
using System.IO;
using Aspose.Diagram;
using System;
namespace Aspose.Diagram.Examples.CSharp.Diagrams
{
    public class RetrieveMasterInfo
    {
        public static void Run()
        {
            // ExStart:RetrieveMasterInfo
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_Master();

            // Call a Diagram class constructor to load the VDX diagram
            Diagram vdxDiagram = new Diagram(dataDir + "RetrieveMasterInfo.vdx");

            foreach (Aspose.Diagram.Master master in vdxDiagram.Masters)
            {
                // Display information about the masters
                Console.WriteLine("\nMaster ID : " + master.ID);
                Console.WriteLine("Master Name : " + master.Name);
            }
            
            Console.ReadLine();
            // ExEnd:RetrieveMasterInfo
        }
    }
}